go-controller/pkg/ovn/pod_selector_address_set.go

```diff
@@ -136,39 +138,42 @@ func (oc *DefaultNetworkController) EnsurePodSelectorAddressSet(podSelector, nam
 	return
 }
 
-func (oc *DefaultNetworkController) DeletePodSelectorAddressSet(addrSetKey, backRef string) error {
-	return oc.podSelectorAddressSets.DoWithLock(addrSetKey, func(key string) error {
-		psAddrSet, found := oc.podSelectorAddressSets.Load(key)
+func (bnc *BaseNetworkController) DeletePodSelectorAddressSet(addrSetKey, backRef string) error {
+	return bnc.podSelectorAddressSets.DoWithLock(addrSetKey, func(key string) error {
+		psAddrSet, found := bnc.podSelectorAddressSets.Load(key)
 		if !found {
 			return nil
 		}
 		delete(psAddrSet.backRefs, backRef)
 		if len(psAddrSet.backRefs) == 0 {
-			err := psAddrSet.destroy(oc)
+			err := psAddrSet.destroy(bnc)
 			if err != nil {
 				// psAddrSet.destroy will set psAddrSet.needsCleanup to true in case of error,
 				// cleanup should be retried later
 				return fmt.Errorf("failed to destroy pod selector address set %s: %v", addrSetKey, err)
 			}
-			oc.podSelectorAddressSets.Delete(key)
+			bnc.podSelectorAddressSets.Delete(key)
 		}
 		return nil
 	})
 }
 
-func (psas *PodSelectorAddressSet) init(oc *DefaultNetworkController) error {
+func (psas *PodSelectorAddressSet) init(bnc *BaseNetworkController) error {
 	// create pod handler resources before starting the handlers
 	if psas.handlerResources == nil {
-		as, err := oc.addressSetFactory.NewAddressSet(psas.addrSetDbIDs, nil)
+		as, err := bnc.addressSetFactory.NewAddressSet(psas.addrSetDbIDs, nil)
 		if err != nil {
 			return err
 		}
+		ipv4Mode, ipv6Mode := bnc.IPMode()
 		psas.handlerResources = &PodSelectorAddrSetHandlerInfo{
 			addressSet:        as,
 			key:               psas.key,
 			podSelector:       psas.podSelector,
 			namespaceSelector: psas.namespaceSelector,
 			namespace:         psas.namespace,
+			ipv4Mode:          ipv4Mode,
+			ipv6Mode:          ipv6Mode,
```

Comment:
these variables are only used to build the slice of the right size, I would say it is better to just remove them from the object, and use empty constructor

Reply:
I am not sure I understand your suggestion. Can you please be more specific?

Reply:
it is different now since we have `netInfo` also. Are we going to be able to get ipMode from it as @jcaamano suggested?

Reply:
ipMode is not part of netInfo, it is attribute of netConfInfo. But @jcaamano is make a change to merge them together. but not in this PR.